src/logging.cpp

```diff
@@ -262,7 +264,43 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
 
     str_prefixed = LogTimestampStr(str_prefixed);
 
-    m_started_new_line = !str.empty() && str[str.size()-1] == '\n';
+    if (m_ratelimiters.find(source_location) == m_ratelimiters.end()) {
+        // We create a new rate limit window for this source location on its first log attempt.
+        m_ratelimiters.emplace(source_location, LogRatelimiter{RATELIMIT_WINDOW_SIZE, RATELIMIT_WINDOW_MAX_BYTES});
+    }
+
+    uint64_t dropped_bytes = m_ratelimiters[source_location].GetDroppedBytes();
```

Comment:
I dropped the non-default constructor since we only ever constructed with `RATELIMIT_WINDOW_SIZE` and `RATELIMIT_WINDOW_MAX_BYTES`. So now a new `LogRateLimiter` will be constructed using the default constructor when `[]` is used the first time.